cmd/csi-snapshotter/main.go

```diff
@@ -63,6 +63,9 @@ var (
 	csiAddress                      = flag.String("csi-address", "/run/csi/socket", "Address of the CSI driver socket.")
 	createSnapshotContentRetryCount = flag.Int("create-snapshotcontent-retrycount", 5, "Number of retries when we create a snapshot content object for a snapshot.")
 	createSnapshotContentInterval   = flag.Duration("create-snapshotcontent-interval", 10*time.Second, "Interval between retries when we create a snapshot content object for a snapshot.")
+	retryIntervalStart              = flag.Duration("retry-interval-start", time.Second, "Initial retry interval of failed snapshot creation. It doubles with each failure, up to retry-interval-max.")
+	retryIntervalMax                = flag.Duration("retry-interval-max", 5*time.Minute, "Maximum retry interval of failed snapshot creation.")
+	failedSnapshotThreshold         = flag.Int("failed-snapshot-threshold", 15, "The maximum number of retries on snapshot failures. Set 0 to retry indefinitely. Default is 15.")
```

Comment:
I don't think we should have a failed snapshot threshold. We should retry forever until user cancels the operation by deleting the VolumeSnapshot object.

Reply:
Removed `failedSnapshotThreshold` and made it retry forever until the snapshot object cannot be found. I add this check in  `updateSnapshot`.